src/main/java/org/jitsi/videobridge/cc/vp8/VP8AdaptiveTrackProjectionContext.java

```diff
@@ -228,6 +238,47 @@ VP8FrameProjection createVP8FrameProjection(
         return nextVP8FrameProjection;
     }
 
+    private int getMaxSequenceNumberOfFrame(@NotNull RawPacket rtpPacket)
+    {
+        long ssrc = rtpPacket.getSSRCAsLong();
+        Map<Long, Integer> frameToMaxSequenceNumberMap
+            = ssrcToFrameToMaxSequenceNumberMap.get(ssrc);
+
+        if (frameToMaxSequenceNumberMap == null)
+        {
+            return -1;
+        }
+
+        return frameToMaxSequenceNumberMap
+            .getOrDefault(rtpPacket.getTimestamp(), -1);
+    }
+
+    private void storeMaxSequenceNumberOfFrame(@NotNull RawPacket rtpPacket)
+    {
+        long ssrc = rtpPacket.getSSRCAsLong();
+        Map<Long, Integer> frameToMaxSequenceNumberMap
+            = ssrcToFrameToMaxSequenceNumberMap
+            .computeIfAbsent(ssrc, k -> new LRUCache<>(5));
+
+        long timestamp = rtpPacket.getTimestamp();
+        int newMaxSequenceNumber = rtpPacket.getSequenceNumber();
+        if (frameToMaxSequenceNumberMap.containsKey(timestamp))
+        {
+            int previousMaxSequenceNumber
+                = frameToMaxSequenceNumberMap.get(timestamp);
+
+            if (RTPUtils.isOlderSequenceNumberThan(
```

Comment:
does this method work correctly if one of the given sequence numbers is -1?

Reply:
I don't know, but that (one of the given sequence numbers being -1 in this particular spot) shouldn't happen, right?

Reply:
ah, well i noticed that `getMaxSequenceNumberOfFrame` returns that by default, but i notice now that this doesn't use that helper.  Why not? also, it looks like other code does use it, which would insert a `-1` value by default if it wasn't found?  so we could get `-1` here?

Reply:
I'm really not sure I follow here :) Other code uses the return value of -1 in different ways, but I don't see the problem with that?

Reply:
well, it's possible one of the sequence numbers we pass here isn't a sequence number at all, but the value `-1` which represents 'we don't have/know the sequence number'.  `RTPUtils.isOlderSequenceNumberThan`, from what I can tell, doesn't support the case where one of the sequence numbers it is given is invalid.  So, for example, if `previousMaxSequenceNumber` was never set (and is therefore -1) we may get inconsistent/incorrect results when trying to figure out if `sequenceNumber` is newer than `-1`

Reply:
@gpolitis i think i'm up for letting the other discussions go, but i worry about the stuff here because it may be a bug or error-prone for someone to accidentally add a bug in the future.  i think we should:

1) use the getMaxSequenceNumberOfFrame helper here which you define and use elsewhere
2) only check `RTPUtils.isOlderSequenceNumberThan` if we definitely get a sequence number (and not -1)
i.e.:
```
int previousMaxSequenceNumber = getMaxSequenceNumberOfFrame(ssrc, timestamp);
if (previousMaxSequenceNumber == -1 || RTPUtils.isOlderSequenceNumberThan(previousMaxSequenceNumber, sequenceNumber) 
{
  updateMaxSequenceNumberOfFrame(ssrc, timestamp, sequenceNumber);
}
```
(`updateMaxSequenceNumberOfFrame` does not currently exist, but seems like it should if we have its getter counterpart `getMaxSequenceNumberOfFrame`?).

Reply:
Thanks for insisting on this, there was a hidden null pointer exception in there (implicitly casting to int from Integer).